src/test/ui/rfc-2008-non-exhaustive/reachable-patterns.rs

```diff
@@ -0,0 +1,64 @@
+// Test that the reachable_patterns lint is triggered correctly.
+
+#![deny(reachable_patterns)]
+
+// aux-build:enums.rs
+extern crate enums;
+// aux-build:structs.rs
+extern crate structs;
+
+use enums::{EmptyNonExhaustiveEnum, NonExhaustiveEnum};
+use structs::{NormalStruct, UnitStruct, TupleStruct, FunctionalRecord};
+
+#[non_exhaustive]
+pub struct Foo {
+    a: u8,
+    b: usize,
+    c: String,
+}
+
+#[non_exhaustive]
+pub enum Bar {
+    A,
+    B,
+    C,
+}
+
+fn main() {
+    let enumeration = Bar::A;
+    let structure = Foo { a: 1, b: 1, c: "hello".to_owned(), };
+
+    match enumeration {
+        Bar::A => {},
+        Bar::B => {},
+        #[deny(reachable)]
+        _ => {}
+    }
+    //~^^^^^^ missing patterns of non_exhaustive type
+    //~| HELP add `Bar::C` to match all reachable patterns
+
+    match enumeration {
+        Bar::A => {},
+        Bar::B => {},
+        _ => {}
+    }
+
+    match NonExhaustiveEnum::Unit {
+        NonExhaustiveEnum::Unit => {},
+        NonExhaustiveEnum::Tuple(_) => {},
+        #[warn(reachable)]
+        _ => {}
+    }
+    //~^^^^^^ missing patterns of non_exhaustive type
+    //~| HELP add `Struct` to match all reachable patterns
+
+    let Foo { a, b, .. } = structure;
+    //~^ ERROR missing patterns of non_exhaustive type
+    //~| HELP add `c` to match all reachable patterns
+
+    let FunctionalRecord { first_field, second_field, .. } = FunctionalRecord::default();
```

Comment:
Given the current implementation, where a `#[{deny,warn}(reachable}]` is required for the lint to trigger at a location (which I'm not sure should be the case, see the other comments), I don't understand why the let-binding pattern matching shouldn't also require that (or if this should only apply to match expressions then why the lint fires here at all).

Reply:
It does just hadn't gotten that far, also see the above comment about the stuct `..` issue. I should probably mark this as a draft PR?

Reply:
It makes sense to keep this as a draft for now.